### ISSUE TITLE
MacOS: current code won't launch unless -conf file specified

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#1300]
The most recent release builds and runs successfully in macOS. Current code builds in macOS, but won't run unless a conf file is specified on the command line.

**Comment by joncampbell123** [comment]
Is this SDL or SDL2?

I'm able to run src/dosbox-x just fine without any command line options.

**Comment by emendelson** [comment]
Sorry - I got the problem wrong. As you say, current code works when run from the command line, but it crashes when double-clicked - and that means it won't work if you put it in the MacOS folder in a Mac app. The most recent release code runs when double-clicked.

I tested this with SDL1, and am building SDL2 now and will edit this comment when it's done.

**Comment by joncampbell123** [comment]
I just compiled the code, made the .app folder, and ran it from the Finder, and it worked.

**Comment by joncampbell123** [comment]
I also ran the binary directly from the Finder, which launched the Terminal and then ran just fine.

**Comment by joncampbell123** [comment]
I'm still on Mojave.

**Comment by emendelson** [comment]
Well, this seems to be a weirdness in the Mac I was using. I have two machines: a 2015 MacBook Air  running Mojave and a 2015 MacBook Pro dual-booting between Mojave and Catalina. The SDL1 code does NOT launch on the MacBook Air running Mojave, but it DOES run correctly on the MacBook Pro running Mojave. Both use the same display resolution. 

I compared the terminal output from both copies. The MacBook Air has "Segmentation Fault: 11" where the MacBook Pro has "VOODOO LFB now at d0000000", and of course the terminal output from the Pro continues. 

So this is pretty clearly not a problem in the code but in my system. Apologies for bandwidth-wasting...

**Comment by Allofich** [comment]
> Is there some obscure change in the VOODOO-related code that might be causing this?

Recently a PR I submitted changed around some iterators, including those in the voodoo files:
https://github.com/joncampbell123/dosbox-x/pull/1294/files

If you can build from source, maybe you could build from just before and after that PR was accepted and see if anything changes.

**Comment by emendelson** [comment]
I’ll build and test later today and will report results.

**Comment by emendelson** [comment]
I spent a lot of time building many versions, and I'm hesitant to report what I found, because it seems physically impossible.

Again, the problem is that, in recent code, the SDL1 build won't launch when double-clicked in the Finder.

On my MacBook Air, the last commit that runs correctly is 039b198:

https://github.com/joncampbell123/dosbox-x/commit/039b198462b36abc05e1d640cfc9b9232f6020e3

The first one that does not work (and all the ones afterwards that I've tried don't work either) is 8298733:

https://github.com/joncampbell123/dosbox-x/commit/8298733352e1efe972da66b8a3b13062f8880749

This seems to be physically impossible, because the only thing changed in that build is some version numbers. But I've tested it a dozen times and can repeat the results. If you like, I can post the executables for testing.

**Comment by Allofich** [comment]
If you change the build number back to the previous one, does the most recent code work on the MacBook Air?

**Comment by joncampbell123** [comment]
DOSBox-X starts up by default with the OpenGL output. Does starting up with the surface output help?

Mac OS X's OpenGL library is a bit unstable and crashy, I've noticed.

**Comment by joncampbell123** [comment]
OpenGL is default because the Cocoa NSWindow bitmap based display has worse performance, especially on Mojave

**Comment by emendelson** [comment]
This makes no sense, and it seems physically impossible, but I've tested it repeatedly. The problem I found in launching SDL1 code occurs when the version number in configure.ac is 0.82.23. If I change it to 0.82.22 or 0.82.24, the problem doesn't occur. It occurred to me that an odd number might be the problem, but 0.82.25 doesn't cause the problem. I know this seems impossible and I wouldn't believe it I hadn't seen it. And as it only occurs on my MacBook Air, not on my MacBook Pro, that makes it even harder to figure out. 

I also tried starting with a dosbox.conf that sets output=surface, but that made no difference. This is very, very strange.

**Comment by joncampbell123** [comment]
@emendelson If you change the version number and then do a full recompile, does it crash?

**Comment by emendelson** [comment]
@joncampbell123 Yes, strange but true. I went back to the current code that I had modified with version number 0.82.25, double-clicked on the executable, and it run. I then changed the version number in configure.ac back to 0.85,23, ran "make clean" and "./build-macosx". The resulting executable gives "Segmentation Fault: 11" when I double-click on it. 

I really, really would not have believed this if I hadn't seen it over and over again.
